### PR TITLE
Add simple span processor

### PR DESCRIPTION
**Pull request by plantfansam**
See: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#simple-processor

### CODE REVIEW

File: lib/opentelemetry/trace/simple_span_processor.lua [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+-- @return              processor
+--------------------------------------------------------------------------------
+function _M.new(exporter)
+    return setmetatable({
+        exporter = exporter,
+    }, mt)
+end
+
+function _M.on_end(self, span)
+    if not span.ctx:is_sampled() or self.closed then
```
Comment (yangxikun) [inline lib/opentelemetry/trace/simple_span_processor.lua:29]: Should we set `self.closed = true` in `shutdown`?

Comment (yangxikun) [inline lib/opentelemetry/trace/simple_span_processor.lua:29]: Should we set `self.closed = true` in `shutdown`?

**Review by yangxikun — CHANGES_REQUESTED** [review]

File: lib/opentelemetry/trace/simple_span_processor.lua (hunk 1 above)
Comment (plantfansam) [reply lib/opentelemetry/trace/simple_span_processor.lua:29]: Yeah, my mistake.

Comment (plantfansam) [reply lib/opentelemetry/trace/simple_span_processor.lua:29]: 0e80104cc40de140a3b26f439e3a3f7591f20af4

**Review by yangxikun — APPROVED** [review]

**Pull request merged**